erigon-lib/state/aggregator.go

```diff
@@ -1727,6 +1727,18 @@ func (a *Aggregator) BuildFilesInBackground(txNum uint64) chan struct{} {
 	return fin
 }
 
+func (ac *AggregatorRoTx) StartingTxNum() uint64 {
```

Comment:
I think we must not think that all domains are equal:
- commitment domain already has disabled history 
- we may have new domains in future
- Historical exec does read: acc, storage, code, receipt. but it doesn’t read commitment domain.  


so, maybe we must accept list of domain names as a parameter.

Reply:
The thing is when we create the historical state reader using  `rpchelper.CreateHistoryStateReader` we don't pass the domains we need for it, just the txNum.

If I make the start tx num calculation function take a list of domain names as a parameter, that means that list needs to be specified by the caller of `rpchelper.CreateHistoryStateReader` which is not very optimal I think.

What I notice is that `HistoryReaderV3` is mainly used as a `StateReader`, and seems to only read from:
- accounts
- storage
- code

Receipts and commitment history don't seem to be accessed.

Do you think it would make sense to consider accounts, storage, and code history in unison (as a single reader), and have separate readers for receipts and commitment? (this I can ignore receipts and commitment history startingTxNum  for this reader)

Reply:
HistoryStateReader - is the "buisness-logic thing" - means only HistoryStateReader knows which domains it using. HistoryStateReader - has only 1 use-case: execution of historical blocks (other places in code can call GetAsOf method directly). And re-execution require only acc/code/storage domains.

`which is not very optimal I think` - what do you mean? 

Don't forget that Commitment history is disabled - even on `archive` mode. 

`and have separate readers for receipts and commitment?` - we already have: `ReceiptGenerator`. If some code need commitment - it can call Get or GetAsOf directly - don't need reader (or need some very-specialized reader). 

Any idea on: what to do in `--prune.mode=minimal`? Here can have 0 history files but some history in DB (1 step).

Reply:
`// return the earliest known txnum in state history (excluding commitment and receipt history)` -  Commitment history is disabled (no history files. no history data in db).